aips/wallet-standard.md

```diff
@@ -0,0 +1,325 @@
+
+In this proposal, we suggest bringing a generalized event-based model communication between a wallet and a dapp to Aptos that eliminates all the above issues.
+
+## Impact
+
+1. Dapp developers
+
+- Familiarize themselves with [the new standard](https://github.com/aptos-labs/wallet-standard/tree/main)
+- Migrate to the [new TypeScript SDK](https://github.com/aptos-labs/aptos-ts-sdk)
+- Support a wallet discoverable function and filter out non-aptos wallets
+
+2. Wallet developers
+
+- Familiarize themselves with [the new standard](https://github.com/aptos-labs/wallet-standard/tree/main)
+- Migrate to the [new TypeScript SDK](https://github.com/aptos-labs/aptos-ts-sdk)
+  - Or hold a conversion layer from the new TypeScript SDK types to the legacy TypeScipt SDK types
+- Register the wallet so it will be discoverable by the dapp
+- Implementation of a AptosWallet class that conforms with the new standard
+
+## Rationale
+
+This [chain agnostic solution](https://github.com/wallet-standard/wallet-standard) has been introduced as a generalized standard for wallets and dapps communication and has already been implemented on [Solana](https://github.com/wallet-standard/wallet-standard) and [Sui](https://docs.sui.io/standards/wallet-standard), and the [Ethereum](https://eips.ethereum.org/EIPS/eip-6963) community recently proposed a similar solution.
+
+## Specification
+
+The [Wallet Standard](https://github.com/aptos-labs/wallet-standard) is a chain-agnostic set of interfaces and conventions that aim to improve how applications interact with injected wallets.
+
+**Standard Features**
+
+A standard feature is a method that must or should be supported and implemented by a wallet.
+Here is a list of the suggested [Aptos features](https://github.com/aptos-labs/wallet-standard/tree/main/src/features)
+
+> a feature marked with `*` is an optional feature
+
+`aptos:connect` method to establish a connection between a dapp and a wallet.
+
+```ts
+// `silent?: boolean` - gives ability to trigger connection without user prompt (for example, for auto-connect)
+// `networkInfo?: NetworkInfo` - defines the network that the dapp will use (shortcut for connect and change network)
+
+connect(silent?: boolean, networkInfo?: NetworkInfo): Promise<UserResponse<AccountInfo>>;
+```
+
+`aptos:disconnect` method to disconnect a connection established between a dapp and a wallet
+
+```ts
+disconnect(): Promise<void>;
+```
+
+`aptos:getAccount` to get the current connected account in the wallet
+
+```ts
+getAccount():Promise<UserResponse<AccountInfo>>
+```
+
+`aptos:getNetwork` to get the current network in the wallet
+
+```ts
+getNetwork(): Promise<UserResponse<NetworkInfo>>;
+```
+
+`aptos:signAndSubmitTransaction` method to sign and submit a transaction using the current connected account in the wallet.
```

Comment:
I think this should be optional.
The same functionality can be achieved by an `aptos:signTransaction` + submitting the transaction using the AptosClient.

The new Aptos Wallet Adapter can easily provide a fallback implementation.

Reply:
I also think a wallet should only supports `sign` functionality. no `signAndSubmitTransaction` or `submitTransaction`. will make this optional for a wallet to implement